### DISCUSSION TITLE
Event that can handle app being terminated

### DISCUSSION THREAD

**Discussion by mikernet** [unoplatform/uno#18822] [General]
Is there any suitable event somewhere that can be used to determine when the application (iOS and Android) is being terminated and perform some actions upon termination? Suspending is not suitable as the app still receives background location updates after being suspended. I need to know when the app is actually terminated (i.e. swiped away, device restarting, etc), so that I can mark the time that location tracking actually stopped in persistent storage.

If not then I suppose an alternative that could work is a background service with a timer on a 1 minute interval that saves the time it fired each time, but that seems like a bit of a heavy approach just to know when the app terminated last.

**Comment by jeromelaban** [comment]
On iOS and Android, you may not be able to reliable know when the App is bein terminated. If the app was suspended, the user or the system may terminate it without the app being resumed.
